vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -692,17 +701,58 @@ func (l *terminationLoggingListener) Accept() (net.Conn, error) {
 
 	select {
 	case <-l.lateStopCh:
-		klog.Warningf("Accepted new connection from %s very late in the graceful termination process (more than 80%% has passed), possibly a sign for a broken load balancer setup.", c.RemoteAddr().String())
-		if swapped := l.lateConnReceived.CAS(false, true); swapped {
-			l.Eventf(corev1.EventTypeWarning, "LateConnections", "The apiserver received connections (e.g. from %q) very late in the graceful termination process, possibly a sign for a broken load balancer setup.", c.RemoteAddr().String())
-		}
+		lateConnectionRemoteAddrsLock.Lock()
+		defer lateConnectionRemoteAddrsLock.Unlock()
+		lateConnectionRemoteAddrs[c.RemoteAddr().String()] = true
```

Comment:
it looks like `RemoteAddr()` can return `nil` value - https://golang.org/src/net/net.go